R/diagnostic_test.R

```diff
@@ -0,0 +1,198 @@
+#' Diagnostic testing
+#'
+#' @description
+#' Run a set of diagnostic tests to check for common issues found when setting
+#' up R on a DfE
+#' system. This includes:
+#'   - Checking for proxy settings in the Git configuration
+#'   - Checking for correct download method used by renv (curl)
+#'
+#' @inheritParams check_proxy_settings
+#'
+#' @return List object of detected parameters
+#' @export
+#'
+#' @examples
+#' diagnostic_test()
+diagnostic_test <- function(
+    clean = FALSE,
+    verbose = FALSE) {
+  results <- c(
+    check_proxy_settings(clean = clean, verbose = verbose),
+    check_github_pat(clean = clean, verbose = verbose),
+    check_renv_download_method(clean = clean, verbose = verbose)
+  )
+  return(results)
+}
+
+#' Check proxy settings
+#'
+#' @description
+#' This script checks for "bad" proxy settings. Prior to the pandemic, analysts
+#' in the DfE would need to add some proxy settings to their GitHub config.
+#' These settings now prevent Git from connecting to remote archives on GitHub
+#' and Azure DevOps if present, so this script identifies and (if clean=TRUE is
+#' set) removes them.
```

Comment:
The old code (https://github.com/dfe-analytical-services/dfeR/commit/3233fe1c20988ccc8dc5c9e607b8ea32c7694bc0#diff-9152ef3c45461a2c8473510069ec582ac76fa432c711c8d9279d32dcceaa2b98L21) set http_proxy and https_proxy as environment variables - I think this function should be checking for those too using `Sys.getenv()`?